app/services/post_scraper.rb

```diff
@@ -1,88 +1,59 @@
-    'peterverse'         => 'Pedro',
-    'curiousdiscoverer'  => 'CuriousDiscoverer',
-    'aestrix'            => 'Aestrix',
-    'unbitwise'          => 'Unbitwise',
-    'erinflight'         => 'ErinFlight',
-    'andaisq'            => 'andaisq',
-    'rockeye-stonetoe'   => 'Rockeye',
-    'rockeye_stonetoe'   => 'Rockeye',
-    'maggie-of-the-owls' => 'MaggieoftheOwls',
-    'maggie_of_the_owls' => 'MaggieoftheOwls', # have both - and _ versions cause Dreamwidth supports both
-    'nemoconsequentiae'  => 'Nemo',
-    'armokgob'           => 'Armok',
-    'timepoof'           => 'Timepoof',
-  }
+class PostScraper < Generic::Service
+  SANDBOX_ID = 3
 
-  attr_accessor :url, :post, :html_doc
+  attr_reader :url
 
-  def initialize(url, board_id=nil, section_id=nil, status=nil, threaded_import=false, console_import=false, subject=nil)
-    @board_id = board_id || SANDBOX_ID
+  def initialize(url, board_id: SANDBOX_ID, section_id: nil, status: Post::STATUS_COMPLETE, threaded: false, console: false, subject: nil)
+    @board_id = board_id
     @section_id = section_id
-    @status = status || Post::STATUS_COMPLETE
-    url += (url.include?('?') ? '&' : '?') + 'style=site' unless url.include?('style=site')
-    url += '&view=flat' unless url.include?('view=flat') || threaded_import
-    @url = url
-    @console_import = console_import
-    @threaded_import = threaded_import # boolean
+    @status = status
+    @console_import = console
+    @threaded_import = threaded # boolean
     @subject = subject
+    @url = clean_url(url)
+    super()
   end
 
-  def scrape!
-    @html_doc = doc_from_url(@url)
+  def scrape(threads=nil)
+    html_doc = doc_from_url(@url)
 
     Post.transaction do
-      import_post_from_doc(@html_doc)
-      import_replies_from_doc(@html_doc)
-      links = page_links
-      links.each_with_index do |link, i|
-        logger.debug "Scraping '#{@post.subject}': page #{i+1}/#{links.count}"
-        doc = doc_from_url(link)
-        import_replies_from_doc(doc)
+      import_post_from_doc(html_doc)
+      if threads.present?
+        threads.each { |thread| evaluate_links(doc_from_url(thread)) }
+      else
+        evaluate_links(html_doc)
       end
       finalize_post_data
     end
+    return if @errors.present?
     GenerateFlatPostJob.perform_later(@post.id)
     @post
   end
+  alias scrape! scrape
 
-  # works as an alternative to scrape! when you want to scrape particular top-level threads of a post sequentially
+  # works as an alternative to scrape when you want to scrape particular top-level threads of a post sequentially
   # "threads" are URL permalinks to the threads to scrape, which it will scrape in the given order
-  def scrape_threads!(threads)
-    raise RuntimeError.new('threaded_import must be true to use scrape_threads!') unless @threaded_import
+  def scrape_threads(threads)
+    @errors.add(:base, 'threaded_import must be true to use scrape_threads') && return unless @threaded_import
```

Comment:
Does `@errors.add` definitely return a truthy value? If not, we won't return. Might be worthwhile turning this into a block instead.